test/projen/projen-struct.test.ts

```diff
@@ -325,6 +325,42 @@ test('can override import locations', () => {
   expect(renderedFile).toContain("import { typescript } from 'banana';");
 });
 
+test('can nest one struct within another', () => {
+  // ARRANGE
+  const project = new TestProject();
+
+  // ACT
+  const base = new ProjenStruct(project, {
+    name: 'MyBaseInterface',
+  });
+
+  const nestedStruct = new ProjenStruct(project, {
+    name: 'MyInterface',
+  });
+  nestedStruct
+    .mixin(Struct.fromFqn('projen.typescript.ProjenrcOptions'))
+    .update('filename', { docs: { default: '.projenRC.ts' } });
+
+  base.nest('projenrcTsOptions', {
+    struct: nestedStruct,
+    docs: {
+      summary: 'My new Summary',
+    },
+  });
```

Comment:
Do I understand this correctly that this is basically a shortcut for (assuming `_base` were public)?

```suggestion
  base.update('projenrcTsOptions', {
    type: {
      fqn: nestedStruct._base.fqn
    },
    docs: {
      summary: 'My new Summary',
    },
  });
```

Reply:
yes, if base was public, I think, it would have worked.

Reply:
Nice. So I think what we should do instead is introducing a new interface:

```ts
export interface NamedTypeReference  {
    /**
     * The fully-qualified-name of the type.
     */
    fqn: FQN;
}
```

And have `Struct` and `ProjenStruct` implement the interface.

Because then the code should become the following, which I find rather nice:
```ts
  base.update('projenrcTsOptions', {
    type: nestedStruct,
    docs: {
      summary: 'My new Summary',
    },
  });
```

What do you think?

Reply:
Thanks for your feedback it took me a while to figure out the implementationa. But after that I agree, it does look nice and easy to use.